controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -75,6 +75,12 @@ var (
 	// NOTE: The following assumes that kubeadm version equals to Kubernetes version.
 	minVerKubeletSystemdDriver = semver.MustParse("1.21.0")
 
+	// Starting from v1.24.0 kubeadm uses "kubelet-config" a ConfigMap name for KubeletConfiguration,
+	// Dropping the X-Y suffix.
+	//
+	// NOTE: The following assumes that kubeadm version equals to Kubernetes version.
+	minVerUnifiedKubeletConfig = semver.MustParse("1.24.0")
```

Comment:
```suggestion
	minVerUnversionedKubeletConfig = semver.MustParse("1.24.0")
```
nit, Should this be called unversioned kubelet config (like the upstream `UnversionedKubeletConfigMap` feature gate)

Reply:
If we change it here, same for the constant